src/Selector/ExtendSelector.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpAT\Selector;
+
+use PhpAT\Parser\AstNode;
+
+class ExtendSelector implements SelectorInterface
+{
+    /**
+     * @var string
+     */
+    private $fqcn;
+    /**
+     * @var AstNode[]
+     */
+    private $astMap;
+
+    public function __construct(string $fqcn)
+    {
+        $this->fqcn = $fqcn;
+    }
+
+    public function getDependencies(): array
+    {
+        return [];
+    }
+
+    public function injectDependencies(array $dependencies): void
+    {
+    }
+
+    /**
+     * @param AstNode[] $astMap
+     */
+    public function setAstMap(array $astMap): void
+    {
+        $this->astMap = $astMap;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function select(): array
+    {
+        foreach ($this->astMap as $astNode) {
+            if ($this->matchesPattern($astNode->getParent(), $this->fqcn)) {
```

Comment:
shouldnt this also support class hierachies over several levels?

Reply:
Yep. This is a feature I have in mind for all the selectors and rule types, with a configuration (enabled by default). And it should report it as an indirect implementation/dependency/etc on validations